packages/store/tests/ensure-store.spec.ts

```diff
@@ -0,0 +1,214 @@
+        CountState.canInheritSelectFn
+      );
+
+      const myCounterMetadata: SelectorMetaDataModel = getSelectorMetadata(
+        MyCounterState.canInheritSelectFn
+      );
+
+      expect(countMetadata.selectorName).toEqual('canInheritSelectFn');
+      expect(myCounterMetadata.selectorName).toEqual('canInheritSelectFn');
+
+      // TODO(splincode): is normal for CountState?
+      expect(countMetadata.containerClass).toEqual(MyCounterState);
+      expect(myCounterMetadata.containerClass).toEqual(MyCounterState);
+
+      expect(countMetadata.originalFn).not.toEqual(CountState.canInheritSelectFn);
+      expect(myCounterMetadata.originalFn).not.toEqual(MyCounterState.canInheritSelectFn);
+
+      expect(extractFunctionBody(countMetadata.originalFn)).toEqual(
+        'function (state) { return state * 2; }'
+      );
+
+      expect(extractFunctionBody(myCounterMetadata.originalFn)).toEqual(
+        'function (state) { return state * 2; }'
+      );
+
+      expect(countMetadata.originalFn!(1)).toEqual(2); // state => state * 2
+      expect(myCounterMetadata.originalFn!(1)).toEqual(2); // state => state * 2
+
+      expect(extractFunctionBody(countMetadata.getSelectorOptions)).toEqual(
+        'function () { return getCustomSelectorOptions(selectorMetaDataClone, getExplicitSelectorOptions()); }'
+      );
+
+      expect(extractFunctionBody(myCounterMetadata.getSelectorOptions)).toEqual(
+        'function () { return getCustomSelectorOptions(selectorMetaDataClone, getExplicitSelectorOptions()); }'
+      );
+
+      expect(countMetadata.getSelectorOptions()).toEqual({ suppressErrors: false });
+      expect(myCounterMetadata.getSelectorOptions()).toEqual({ suppressErrors: false });
+
+      expect(countMetadata.selectFromAppState).toEqual(
+        getSelectorFn(CountState.canInheritSelectFn)
+      );
+
+      expect(myCounterMetadata.selectFromAppState).toEqual(
+        getSelectorFn(MyCounterState.canInheritSelectFn)
+      );
+
+      // TODO(splincode): is normal for CountState, MyCounterState?
+      expect(getSelectorFn(CountState.canInheritSelectFn)(0)).toEqual(NaN);
+      expect(getSelectorFn(MyCounterState.canInheritSelectFn)(0)).toEqual(NaN);
+    });
+
+    it('should get the selector meta data from the SuperCountState.canInheritSelectFn', () => {
+      try {
+        @State({
+          name: 'superCount',
+          defaults: 0
+        })
+        class SuperCountState extends MyCounterState {
+          @Selector()
+          public static canInheritSelectFn(state: number): number {
+            return super.canInheritSelectFn(state) + 1;
+          }
+        }
+
+        const metadata: SelectorMetaDataModel = getSelectorMetadata(
+          SuperCountState.canInheritSelectFn
+        );
+
+        expect(metadata.containerClass).toEqual(SuperCountState);
+      } catch (e) {
+        // TODO(splincode): is normal for SuperCountState?
+        expect(extractFunctionBody(e.message)).toEqual(
```

Comment:
Could you create a local variable and assign it in the `catch` block:
```ts
let error: Error | null = null;

try {
  ...
} catch (e) {
  error = e;
}

expect(error).toBeDefined();
expect(extractFunctionBody....)...
```

Reply:
done